metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/v3/OpenAPIV3Generator.java

```diff
@@ -1344,6 +1363,131 @@ private static Schema buildEntitySchema(
         .properties(properties);
   }
 
+  private static Schema buildSortSchema() {
+    // SortCriterion (inlined)
+    Schema sortCriterion =
+        newSchema()
+            .type(TYPE_OBJECT)
+            .description("Sort criterion: field + direction.")
+            .addProperties(
+                "field", newSchema().type(TYPE_STRING).description("Field name to sort by."))
+            .addProperties(
+                "order",
+                newSchema()
+                    .type(TYPE_STRING)
+                    ._enum(Arrays.asList("ASCENDING", "DESCENDING"))
+                    ._default("ASCENDING")
+                    .description("Sort order (default ASCENDING)."))
+            .addRequiredItem("field")
+            .addRequiredItem("order");
+
+    // Sort (top-level)
+    return newSchema()
+        .type(TYPE_OBJECT)
+        .description("Sort specification made up of one or more SortCriterion in evaluation order.")
+        .addProperties(
+            "sortCriteria",
+            newSchema()
+                .type(TYPE_ARRAY)
+                .description("List of sort criteria; applied in order.")
+                .items(sortCriterion)
+                .minItems(1))
+        .addRequiredItem("sortCriteria");
+  }
+
+  private static Schema buildFilterSchema() {
+    // Criterion schema
+    Schema criterion =
+        newSchema()
+            .type("object")
+            .description("A single field condition.")
+            .addProperties(
+                "field", newSchema().type(TYPE_STRING).description("The name of the field"))
+            .addProperties(
+                "value",
+                newSchema()
+                    .type(TYPE_STRING)
+                    .description("Single value. Mutually exclusive with `values`."))
+            .addProperties(
+                "values",
+                newSchema()
+                    .type(TYPE_ARRAY)
+                    .items(newSchema().type(TYPE_STRING))
+                    .minItems(1)
+                    .description(
+                        "List of values (match any). If set, `value` must not be provided."))
+            .addProperties(
+                "condition",
+                newSchema()
+                    .type(TYPE_STRING)
+                    ._enum(Arrays.asList("EQUAL", "STARTS_WITH", "ENDS_WITH", "EXISTS", "IN"))
+                    ._default("EQUAL")
+                    .description("The condition for the criterion."))
+            .addProperties(
+                "negated",
+                newSchema()._default(false).description("Whether the condition should be negated."))
+            .addRequiredItem("field")
+            .addRequiredItem("condition");
+
+    // oneOf branches for Criterion
+    Schema<?> branchValue =
+        newSchema()
+            .type(TYPE_OBJECT)
+            .addRequiredItem("value")
```

Comment:
We should not expose `value` in the API, this has been deprecated and only exists due to internal requirements around backwards/legacy compatibility. Parts of the codebase will refuse to consider `value`.

Reply:
Also having both `value` and `values` doesn't really make sense since an array of 1 element seems perfectly reasonable.